tests/blotter/test_core.py

```diff
@@ -0,0 +1,173 @@
+        assert_equal(self.blotters, mappingproxy({}))
+
+    def test_load_not_registered(self):
+        msg = "no blotter class registered as 'ayy-lmao', options are: []"
+        with assert_raises_str(ValueError, msg):
+            self.load('ayy-lmao')
+
+        # register in reverse order to test the sorting of the options
+        self.register('c', SimulatedBlotter)
+        self.register('b', SimulatedBlotter)
+        self.register('a', SimulatedBlotter)
+
+        msg = (
+            "no blotter class registered as 'ayy-lmao', options are: "
+            "['a', 'b', 'c']"
+        )
+        with assert_raises_str(ValueError, msg):
+            self.load('ayy-lmao')
+
+    def test_register_decorator(self):
+
+        @self.register('ayy-lmao')
+        class ProperDummyBlotter(Blotter):
+
+            def order(self, asset, amount, style, order_id=None):
+                pass
+
+            def cancel(self, order_id, relay_status=True):
+                pass
+
+            def cancel_all_orders_for_asset(self, asset, warn=False,
+                                            relay_status=True):
+                pass
+
+            def execute_cancel_policy(self, event):
+                pass
+
+            def reject(self, order_id, reason=''):
+                pass
+
+            def hold(self, order_id, reason=''):
+                pass
+
+            def process_splits(self, splits):
+                pass
+
+            def get_transactions(self, bar_data):
+                pass
+
+            def prune_orders(self, closed_orders):
+                pass
+
+        expected_blotters = mappingproxy({'ayy-lmao': ProperDummyBlotter})
+        assert_equal(self.blotters, expected_blotters)
+        assert_is(self.load('ayy-lmao'), ProperDummyBlotter)
+
+        msg = "blotter class 'ayy-lmao' is already registered"
+        with assert_raises_str(ValueError, msg):
+            @self.register('ayy-lmao')
+            class Fake(object):
+                pass
+
+        msg = "The class specified is not a subclass of Blotter"
+        with assert_raises_str(TypeError, msg):
+            @self.register('something-different')
+            class ImproperDummyBlotter(object):
+                pass
+
+        # ensure that the failed registration didn't break the previously
+        # registered blotter
+        assert_equal(self.blotters, expected_blotters)
+        assert_is(self.load('ayy-lmao'), ProperDummyBlotter)
+
+        self.unregister('ayy-lmao')
+        assert_equal(self.blotters, mappingproxy({}))
+
+        msg = "no blotter class registered as 'ayy-lmao', options are: []"
+        with assert_raises_str(ValueError, msg):
+            self.load('ayy-lmao')
+
+        msg = "blotter class 'ayy-lmao' was not already registered"
+        with assert_raises_str(ValueError, msg):
+            self.unregister('ayy-lmao')
+
+    def test_register_non_decorator(self):
+
+        class ProperDummyBlotter(Blotter):
```

Comment:
To reduce the number of things that need to change if we change the blotter, can we just use an empty subclass of `SimulatedBlotter` here?

Reply:
Sure